src/pty.rs

```diff
@@ -0,0 +1,128 @@
+//! Create master and slave virtual pseudo-terminals (PTTYs)
+
+use std::ffi::CStr;
+use std::os::unix::io::RawFd;
+
+use libc;
+
+use {Error, fcntl, Result};
+
+/// Grant access to a slave pseudoterminal (see
+/// [grantpt(3)](http://man7.org/linux/man-pages/man3/grantpt.3.html))
+///
+/// `grantpt()` changezs the mode and owner of the slave pseudoterminal device corresponding to the
+/// master pseudoterminal referred to by `fd`. This is a necessary step towards opening the slave.
+#[inline]
+pub fn grantpt(fd: RawFd) -> Result<()> {
```

Comment:
Yeah, I agree, maybe just `AsRawFd` is a better trait instead? But no, nothing else really deals with this. I have some code in my serialport-rs library [here](https://gitlab.com/susurrus/serialport-rs/blob/master/src/posix/tty.rs#L239) that deals with this same thing, and that's like the only other thing working with safe ptty abstractions.